### DISCUSSION TITLE
Blazor Support - rev.1

### DISCUSSION THREAD

**Discussion by adrianhall** [Azure/azure-mobile-apps#538] [Show and tell]
Hey folks,

I decided to play around with Azure Mobile Apps + Blazor WASM.  I've created a sample, which I merged into the samples folder earlier today.

Short version of what is working / what is not working:
1. You have to set an InstallationId in the DatasyncClientOptions for anything to work.  Issues #537 filed to fix this; I know what needs to be done, but it needs to be done in a way that doesn't break everyone else, so I'm going to run through all the tutorials from beginning to end before I declare it ready for release.
2. Authentication is a little wierd.  You basically can't do the "automatically authenticate" thing that can be done in mobile apps; instead you have to deal with authentication outside of the main flow, then provide a generic authentication token provider that just reads your authentication token.  Relative difficulty - middling (non-obvious).
3. Offline is completely a non-starter with SQLite.   This was unsurprising.  I'm going to see if I can make it all work with an IndexedDb through JSInterop.

Anyone working on exciting stuff over the Christmas period?

**Comment by fileman** [comment]
Hi Adrian,
there is a reason to not use something like https://github.com/JeremyLikness/SqliteWasmHelper ?

**Reply by adrianhall** [reply]
Potentially - I'll have to take a look.  It looks promising.
